<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>"CITEViz: Classifying Cell Clusters in CITE-Seq Data using the Flow Cytometry Gating Workflow" • CITEViz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="" citeviz: classifying cell clusters in cite-seq data using the flow cytometry gating workflow>
<meta property="og:description" content="CITEViz is an RShiny app that allows users to classify cell types in CITE-Seq data via cellular surface protein levels. Similar to flow cytometry, users can physically draw gates and filter on a mixture of cells, and the selected cells are immediately highlighted in dimensional reduction space (e.g. UMAP, tSNE, PCA). This workflow provides rapid feedback to the users to help classify cell types. Additional features of CITEViz includes assessment of quality control metrics, feature visualization (feature expression and co-expression), and plots of cell clusters (2D and optionally 3D). The input of CITEViz is a pre-processed Seurat/SingleCelExperiment object with assays containing normalized RNA/SCT (single-cell transform) and ADT (antibody-derived tag) count data, as well as data for at least one dimensional reduction (e.g. UMAP, PCA, etc.).">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CITEViz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/CITEViz.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/maxsonBraunLab/CITEViz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="citeviz">CITEViz<a class="anchor" aria-label="anchor" href="#citeviz"></a>
</h1></div>
<!-- badges: start -->

<p><img src="articles/citeviz-logo-light-cropped.png" alt="drawing" width="200"></p>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>The rapid advancement of new genomic sequencing technology has enabled the development of multi-omic single-cell sequencing assays. These assays profile multiple modalities in the same cell and can often yield new insights not revealed with a single modality. For example, CITE-Seq (Cellular Indexing of Transcriptomes and Epitopes by Sequencing) simultaneously profiles the single-cell RNA transcriptome and the surface protein expression. The extra dimension of surface protein markers can be used to further identify cell clusters – an essential step for downstream analyses and interpretation. Additionally, multi-dimensional datasets like CITE-Seq require nuanced visualization methods to accurately assess the data. To facilitate cell cluster classification and visualization in CITE-Seq, we developed CITEViz.</p>
</div>
<div class="section level2">
<h2 id="about-citeviz">About CITEViz<a class="anchor" aria-label="anchor" href="#about-citeviz"></a>
</h2>
<p>CITEViz is a single-cell visualization platform with a custom module that replicates the interactive flow-cytometry gating workflow. With CITEViz, users can investigate CITE-Seq specific quality control (QC) metrics, view multi-omic co-expression feature plots, and classify cell clusters by iteratively gating on the abundance of cell surface markers. CITEViz was developed to make multi-modal single-cell analysis accessible to a wide variety of biologists, with the aim to discover new insights into their data and to facilitate novel hypothesis generation. If you use our package, please cite our paper: <a href="https://www.biorxiv.org/content/10.1101/2022.05.15.491411v1" class="external-link">CITE-Viz: Replicating the Interactive Flow Cytometry Workflow in CITE-Seq</a></p>
</div>
<div class="section level2">
<h2 id="installation--usage">Installation &amp; Usage<a class="anchor" aria-label="anchor" href="#installation--usage"></a>
</h2>
<p>We recommend installing CITEViz through the RStudio IDE using the following commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"maxsonBraunLab/CITEViz"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CITEViz</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/run_app.html">run_app</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>CITEViz installation is currently facilitated with devtools, and the development team is working hard to submit the package to Bioconductor 3.16 with a release schedule of October 2022.</p>
<div class="section level3">
<h3 id="feature-1-forward-gates">Feature 1: Forward-Gates<a class="anchor" aria-label="anchor" href="#feature-1-forward-gates"></a>
</h3>
<p>Arguably the most important feature for CITEViz, users can gate on a set of features (ADT, RNA, etc.) to subset cells in the dimensional reduction space (e.g. UMAP, PCA, tSNE). Users can use one or multiple layers of gates to enrich for specific cell populations justl like flow cytometry. In the following example, we demonstrate a 2-layer gate to get CD11b-1 CD45-1 cells (mixture of myeloid and lymphoid cells) and CD8+ CD4- cells to get CD8 T cells:</p>
<ol style="list-style-type: decimal">
<li>Select cells with relatively low levels of CD11b-1 and CD45-1.</li>
</ol>
<div class="figure">
<img src="articles/gating_CD8_T_Cells.png" alt=""><p class="caption">forward-gate1</p>
</div>
<ol start="2" style="list-style-type: decimal">
<li>From the cells in the first gate, select cells with low levels of CD4 and high levels of CD8.</li>
</ol>
<div class="figure">
<img src="articles/gating_CD8_T_Cells2.png" alt=""><p class="caption">forward-gate2</p>
</div>
</div>
<div class="section level3">
<h3 id="feature-2-back-gates">Feature 2: Back-Gates<a class="anchor" aria-label="anchor" href="#feature-2-back-gates"></a>
</h3>
<p>Unique to CITEViz, users can ‘back-gate’ on a selection of cells, and highlight them in feature space. We believe this feature can highlight cells in feature space from a “labels-first” or “top-down” workflow. In the following example, we back-gate on a selection of B-cells:</p>
<div class="figure">
<img src="articles/backgate-b-cells.png" alt=""><p class="caption">back-gate</p>
</div>
</div>
<div class="section level3">
<h3 id="feature-3-quality-control">Feature 3: Quality Control<a class="anchor" aria-label="anchor" href="#feature-3-quality-control"></a>
</h3>
<p>A set of quality control metrics can be assessed in CITEViz, split by any categorical metadata in the input Seurat object. This feature helps users identify potential faulty samples or batch effect. In the following example, CITEViz clearly displayed a difference in the Number of Unique ADTs between patient donors P1-P4 and P5-P8.</p>
<div class="figure">
<img src="articles/qc-unique-adt.png" alt=""><p class="caption">quality-control</p>
</div>
</div>
<div class="section level3">
<h3 id="feature-4-feature-co-expression">Feature 4: Feature (co-)Expression<a class="anchor" aria-label="anchor" href="#feature-4-feature-co-expression"></a>
</h3>
<p>Co-Expression of features greatly facilitates a holistic view of single-cell multi-omic datasets. For example, this co-expression plot of CD14 and CD16 protein expression show cell population heterogeneity in monocytes in the left cluster. Most monocytes have high CD14 cells and moderate CD16, except for a specific population at the bottom dip that have high CD16 and low CD14.</p>
<div class="figure">
<img src="articles/co-expression.png" alt=""><p class="caption">co-expression</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Andreatta M, Berenstein AJ, Carmona SJ. scGate: marker-based purification of cell types from heterogeneous single-cell RNA-seq datasets. :3.</p>
<p>Hao Y, Hao S, Andersen-Nissen E, Mauck WM, Zheng S, Butler A, Lee MJ, Wilk AJ, Darby C, Zager M, et al. 2021. Integrated analysis of multimodal single-cell data. Cell. 184(13):3573-3587.e29. <a href="doi:10.1016/j.cell.2021.04.048" class="uri">doi:10.1016/j.cell.2021.04.048</a>.</p>
<p>Jagla B, Libri V, Chica C, Rouilly V, Mella S, Puceat M, Hasan M. 2021. SCHNAPPs - Single Cell sHiNy APPlication(s). Journal of Immunological Methods. 499:113176. <a href="doi:10.1016/j.jim.2021.113176" class="uri">doi:10.1016/j.jim.2021.113176</a>.</p>
<p>Macosko EZ, Basu A, Satija R, Nemesh J, Shekhar K, Goldman M, Tirosh I, Bialas AR, Kamitaki N, Martersteck EM, et al. 2015. Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets. Cell. 161(5):1202–1214. <a href="doi:10.1016/j.cell.2015.05.002" class="uri">doi:10.1016/j.cell.2015.05.002</a>.</p>
<p>Ouyang JF, Kamaraj US, Cao EY, Rackham OJL. 2021. ShinyCell: simple and sharable visualization of single-cell gene expression data. Mathelier A, editor. Bioinformatics. 37(19):3374–3376. <a href="doi:10.1093/bioinformatics/btab209" class="uri">doi:10.1093/bioinformatics/btab209</a>.</p>
<p>Pont F, Tosolini M, Gao Q, Perrier M, Madrid-Mencía M, Huang TS, Neuvial P, Ayyoub M, Nazor K, Fournié J-J. 2020. Single-Cell Virtual Cytometer allows user-friendly and versatile analysis and visualization of multimodal single cell RNAseq datasets. NAR Genomics and Bioinformatics. 2(2):lqaa025. <a href="doi:10.1093/nargab/lqaa025" class="uri">doi:10.1093/nargab/lqaa025</a>.</p>
<p>Stoeckius M, Hafemeister C, Stephenson W, Houck-Loomis B, Chattopadhyay PK, Swerdlow H, Satija R, Smibert P. 2017. Simultaneous epitope and transcriptome measurement in single cells. Nat Methods. 14(9):865–868. <a href="doi:10.1038/nmeth.4380" class="uri">doi:10.1038/nmeth.4380</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/maxsonBraunLab/CITEViz/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/maxsonBraunLab/CITEViz/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CITEViz</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Garth Kong <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0001-5262-5406" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Thai Nguyen <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8988-4003" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Wesley Rosales <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-6381-9844" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Anjali Panikar <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-5740-807X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>John Cheney <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-6194-2745" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/maxsonBraunLab/CITEViz/actions/workflows/rcmdcheck.yaml" class="external-link"><img src="https://github.com/maxsonBraunLab/CITEViz/actions/workflows/rcmdcheck.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Garth Kong, Thai Nguyen, Wesley Rosales, Anjali Panikar, John Cheney.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
