<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CITEViz: Interactively Classify Cell Populations in CITE-Seq via a Flow Cytometry-like Gating Workflow using R-Shiny • CITEViz</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CITEViz: Interactively Classify Cell Populations in CITE-Seq via a Flow Cytometry-like Gating Workflow using R-Shiny">
<meta property="og:description" content="CITEViz">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CITEViz</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CITEViz.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/maxsonBraunLab/CITEViz/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CITEViz: Interactively Classify Cell Populations
in CITE-Seq via a Flow Cytometry-like Gating Workflow using R-Shiny</h1>
                        <h4 data-toc-skip class="author">Garth Kong</h4>
            <address class="author_afil">
      Division of Oncologic Science, Knight Cancer Institute, Oregon
Health and Science
University<br><h4 data-toc-skip class="author">Thai
Nguyen</h4>
            <address class="author_afil">
      Division of Oncologic Science, Knight Cancer Institute, Oregon
Health and Science
University<br><h4 data-toc-skip class="author">Wesley
Rosales</h4>
            <address class="author_afil">
      Earle A. Chiles Research
Institute<br><h4 data-toc-skip class="author">Anjali
Panikar</h4>
            <address class="author_afil">
      Bioinformatics and Genomics Masters Program, University of
Oregon<br><h4 data-toc-skip class="author">John
Cheney</h4>
            <address class="author_afil">
      Bioinformatics and Genomics Masters Program, University of
Oregon<br><h4 data-toc-skip class="author">Brittany
Curtiss</h4>
            <address class="author_afil">
      Knight Cancer Institute, Oregon Health and Science
University<br><h4 data-toc-skip class="author">Sarah
Carratt</h4>
            <address class="author_afil">
      Knight Cancer Institute, Oregon Health and Science
University<br><h4 data-toc-skip class="author">Theodore
Braun</h4>
            <address class="author_afil">
      Knight Cancer Institute, Oregon Health and Science
University<br><h4 data-toc-skip class="author">Julia
Maxson</h4>
            <address class="author_afil">
      Knight Cancer Institute, Oregon Health and Science
University<br><small class="dont-index">Source: <a href="https://github.com/maxsonBraunLab/CITEViz/blob/HEAD/vignettes/CITEViz.Rmd" class="external-link"><code>vignettes/CITEViz.Rmd</code></a></small>
      <div class="hidden name"><code>CITEViz.Rmd</code></div>

    </address>
</address>
</address>
</address>
</address>
</address>
</address>
</address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><img src="citeviz-logo-light-cropped.png" width="35%" height="35%" style="display: block; margin: auto auto auto 0;"></p>
<p>CITEViz is an R-Shiny program that allows users to classify cell
clusters in CITE-Seq data by iteratively filtering the surface protein
expression similar to flow cytometry. The most unique feature of CITEViz
is the ability to physically gate cells, and the selected cells are
immediately highlighted in dimension reduction space (e.g. UMAP, tSNE,
PCA). This workflow provides rapid feedback to the users to assess cell
cluster identity. Without CITEViz, the alternative workflow is to
manually define and “jitter” gate boundaries (define four cutoff
points), obtain the selected cell barcodes, re-plot the cells using a
new set of antibody features, and re-filter the cells ad infinitum. By
drawing physical gates to filter cells like in flow cytometry, CITEViz
allows rapid exploration of cell clusters in CITE-Seq datasets.</p>
<p>The Interactive SummarizedExperiment Explorer (iSEE) is a
Bioconductor package that shares some functionality with CITEViz. iSEE
allows users to interactively plot SingleCellExperiment data (RNA-Seq,
scRNA-Seq, CyTOF, etc.) with limited preprocessing steps and gating
functionality. CITEViz is intended for users who want to apply a flow
cytometry-like gating workflow to their CITE-Seq data. CITEViz is
intended to be used towards the end of of their CITE-Seq analysis
(including quality control, feature selection, dimension reduction)
performed using Seurat. Both tools can accept data in
SingleCellExperiment format and visualize basic quality control
information.</p>
<p>The motivation to develop CITEViz was to bring bench and
computational biologists together to analyze CITE-Seq data. In our
experience, CITEViz helps leverage the flow cytometry background of
bench scientists to quickly and reliably classify cell clusters, which
can then be further analyzed by computational scientists. To serve an
audience of bench and computational biologists who are familiar with R
the programming language, we submitted CITEViz to the BioConductor
repository.</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Make sure you have R version <strong>4.2.0</strong> or later
installed. We recommend installing CITEViz through the RStudio IDE using
the following commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"CITEViz"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-input-format">Data Input Format<a class="anchor" aria-label="anchor" href="#data-input-format"></a>
</h3>
<p>CITEViz accepts files in the <strong>RDS (.rds) format</strong>. The
preferred RDS file should include a <a href="https://satijalab.org/seurat/" class="external-link">Seurat</a> object or a <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a>
object. If a list of a single <code>Seurat</code> object is used, only
the object labeled “integrated” will be used.</p>
<p>The input <code>Seurat</code> or <code>SingleCellExperiment</code>
object must contain cell embeddings data for <strong>at least one
dimensional reduction method</strong> (e.g. PCA, UMAP, tSNE, etc.). If
the object contains data for more than one reduction, each reduction
must have a unique name. For a <code>SingleCellExperiment</code> object,
the main experiment and any alternative experiments in the
<code>SingleCellExperiment</code> object must contain assay count data
stored as <strong>“logcounts”</strong> (log-transformed, normalized
counts) or <strong>“normcounts”</strong> (normalized counts). Please
refer to the <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a>
reference manual on Bioconductor for help on how to determine the types
of count data present in your SCE object. By default, CITEViz will use
the logcounts for plotting expression data. If logcounts are not
available, then CITEViz will use normcounts. If using a
<code>SingleCellExperiment</code> object that was generated from
Seurat-processed data or converted from a <code>Seurat</code> object,
then the logcounts in the <code>SingleCellExperiment</code> object are
analogous to the counts in the “data” slot for a given assay in a
<code>Seurat</code> object.</p>
<div class="section level4">
<h4 id="sample-data">Sample Data<a class="anchor" aria-label="anchor" href="#sample-data"></a>
</h4>
<p>In this vignette, we used a downsampled version of the PBMC dataset
from <a href="https://doi.org/10.1016/j.cell.2021.04.048" class="external-link">Hao et.
al. 2021</a>. The sample PBMC dataset for this vignette can be
downloaded as a 10K cell subset from Google Drive <a href="https://drive.google.com/drive/folders/1rdXWVMkAshsEM9C-xtixLMsRO4AK3Lgf" class="external-link">here</a>
(<code>small_pbmc.rds</code>), or downloaded as a 2.5K cell subset from
the Bioconductor <a href="https://bioconductor.org/packages/release/bioc/html/ExperimentHub.html" class="external-link">ExperimentHub</a>,
as shown below.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create EH instance and query it</span></span>
<span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">query_results</span> <span class="op">&lt;-</span> <span class="fu">query</span><span class="op">(</span><span class="va">eh</span>, <span class="st">"CITEVizTestData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get sample PBMC CITE-Seq dataset in one of two possible formats</span></span>
<span><span class="va">file_seurat</span> <span class="op">&lt;-</span> <span class="va">query_results</span><span class="op">[[</span><span class="st">"EH7739"</span><span class="op">]</span><span class="op">]</span> <span class="co"># in Seurat format</span></span>
<span><span class="va">file_sce</span> <span class="op">&lt;-</span> <span class="va">query_results</span><span class="op">[[</span><span class="st">"EH7740"</span><span class="op">]</span><span class="op">]</span> <span class="co"># in SingleCellExperiment format</span></span>
<span></span>
<span><span class="co"># Export dataset to an external RDS file in your current working directory</span></span>
<span></span>
<span><span class="co"># Export data as a Seurat object:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">file_seurat</span>, <span class="st">"pbmc-cite-seq-2500-seurat.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Export data as a SingleCellExperiment object:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">file_sce</span>, <span class="st">"pbmc-cite-seq-2500-sce.rds"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="running-citeviz">Running CITEViz<a class="anchor" aria-label="anchor" href="#running-citeviz"></a>
</h3>
<p>To run CITEViz, use the following commands:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://maxsonbraunlab.github.io/CITEViz/" class="external-link">CITEViz</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># run CITEViz</span></span>
<span><span class="fu"><a href="../reference/run_app.html">run_app</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="uploading-data-into-citeviz">Uploading Data into CITEViz<a class="anchor" aria-label="anchor" href="#uploading-data-into-citeviz"></a>
</h4>
<p>After calling <code><a href="../reference/run_app.html">run_app()</a></code> to start CITEViz, a file can be
uploaded using the file upload box at the top of any page in CITEViz.
After clicking the file upload box, a file explorer window containing
files in the user’s local file system will open. The user can select an
RDS file containing CITE-seq data to upload, and the same dataset will
be retained for exploration across all tabs.</p>
<p><img src="file_upload_box.png" width="85%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="exporting-data-from-citeviz">Exporting Data from CITEViz<a class="anchor" aria-label="anchor" href="#exporting-data-from-citeviz"></a>
</h4>
<p>CITEViz uses interactive data visualization and exploration packages
such as <a href="https://plotly-r.com/" class="external-link">Plotly</a> and <a href="https://rstudio.github.io/DT/" class="external-link">DT</a>. <code>Plotly</code> plots
can be exported as PNG files, and <code>DT</code> datatables can be
copied, printed, or exported as files for downstream analysis.</p>
<p>After gating for cell populations of interest (see <a href="#gating">Gating</a> section of this vignette), the metadata for
each gate is saved in an internal Gate object. These metadata include
cell barcodes that can facilitate downstream analysis such as
differential gene expression. To download this gating data as a list of
gates in RDS format, the user can click the “List (.rds)” button at the
bottom of the gating page.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<p>The first step of analysis is to assess the quality of sequencing
data. Here we provide QC plots that display data for common metrics such
as gene or antibody-derived tag (ADT) counts per assay, number of unique
ADTs, and mitochondrial expression, which can be visualized by any
categorical metadata in the user’s <code>Seurat</code> or
<code>SingleCellExperiment</code> object. Dotted lines in each QC plot
represent the values at which 50%, 75%, and 95% of the data falls at or
below that value.</p>
<p><img src="qc-unique-adt.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="clustering">Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h2>
<p>The Clustering page allows the user to view cell clusters in two- and
three-dimensional space. These clusters can be colored by any
categorical metadata, and the user can select dimensionality reductions
(e.g., UMAP, PCA, etc.) to view from a dropdown menu.</p>
<p><img src="clustering_page.png" width="100%" style="display: block; margin: auto;"></p>
<p>When the user’s cursor hovers over the 2D reduction plot, a <a href="https://plotly-r.com/" class="external-link">Plotly</a> toolbox with labeled options
will appear. From this toolbox, the user can zoom, pan, download, and
reset a plot by selecting an option. The user can also use the box or
lasso selection tool to select specific cells in a plot. The metadata
for selected cells appears in the scrollable, interactive data table
below the plots, and the user can print or copy this data to their
clipboard.</p>
<p><img src="clustering_plotly_toolbox.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="feature-expression">Feature Expression<a class="anchor" aria-label="anchor" href="#feature-expression"></a>
</h2>
<p>CITEViz supports RNA and ADT feature expression visualizations in
both one-dimensional and two-dimensional formats. On the feature
expression tabs of CITEViz, cell clusters are displayed in a dimensional
reduction plot. These cell clusters can be colored by expression levels
of selected RNA and/or ADT features. Similar to the clustering page, the
user can select the type of dimensional reduction to view.</p>
<div class="section level3">
<h3 id="one-dimensional-1d-feature-expression">One-Dimensional (1D) Feature Expression<a class="anchor" aria-label="anchor" href="#one-dimensional-1d-feature-expression"></a>
</h3>
<p>In 1D feature expression, cells in a dimensional reduction plot are
colored by expression levels for one RNA or ADT feature. The user can
select a specific RNA or ADT feature from a dropdown menu.</p>
<p><img src="feature_expression_CD8.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="two-dimensional-2d-feature-expression">Two-Dimensional (2D) Feature Expression<a class="anchor" aria-label="anchor" href="#two-dimensional-2d-feature-expression"></a>
</h3>
<p>Co-expression of features greatly facilitates a holistic view of
single-cell multi-omic datasets. In 2D feature expression, cells in a
dimensional reduction plot are colored by expression levels for two gene
and/or ADT features simultaneously. The user can select specific gene
and/or ADT features from dropdown menus. CITEViz can visualize
co-expression of features from the same assay (i.e. two genes or two
ADTs), as shown below, or two features from different assays (i.e. one
gene and one ADT).</p>
<p><img src="feature_expression_CD14_CD16.png" width="100%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="gating">Gating<a class="anchor" aria-label="anchor" href="#gating"></a>
</h2>
<div class="section level3">
<h3 id="forward-gating">Forward Gating<a class="anchor" aria-label="anchor" href="#forward-gating"></a>
</h3>
<p>A key feature of CITEViz is that users can iteratively gate (subset)
cells using antibody markers, and the selected cells are immediately
highlighted in the dimensional reduction space (e.g. UMAP, PCA, tSNE,
etc.). Users can subset cells using one or more gates to explore
specific cell populations similar to flow cytometry.</p>
<div class="section level4">
<h4 id="general-workflow">General Workflow<a class="anchor" aria-label="anchor" href="#general-workflow"></a>
</h4>
<ol style="list-style-type: decimal">
<li>
<strong>Select plot settings:</strong> From dropdown menus in the
sidebar panel, the user can select an assay (e.g. ADT) and two assay
features (e.g. CD4, CD8) for which to plot normalized expression levels.
The user can also select a reduction (e.g. UMAP) for which to plot
dimensional reduction embeddings, and can color the cells in the
reduction plot by a discrete metadata category (e.g. cell type).</li>
</ol>
<p><img src="gating_dropdown_menus.png" width="50%" style="display: block; margin: auto;"></p>
<ol start="2" style="list-style-type: decimal">
<li>
<strong>Select cells:</strong> When the user’s cursor hovers over
each plot on the gating page, a <a href="https://plotly-r.com/" class="external-link">Plotly</a> toolbox with labeled options
will appear. From this toolbox, the user can choose a cell selection
tool, zoom, pan, download, or reset a plot by selecting an option. The
user can use the box or lasso selection tool to select specific cells in
the feature scatter plot. When cells in the feature scatter plot are
selected, the corresponding cells in the reduction plot will be
highlighted.</li>
<li>
<strong>Label the selected cells:</strong> To input a name for a
subset of cells to be gated, use the text-input field of the sidebar
panel prior to clicking the gate button.</li>
<li>
<strong>Click the gate button:</strong> After gating a subset of
cells, information about the newly-created gate will be displayed in a
scrollable, interactive data table below the plots.</li>
<li>
<strong>Label the selected cells:</strong> To input a name for a
subset of cells to be gated, use the text-input field of the sidebar
panel prior to clicking the gate button.</li>
<li>
<strong>Click the gate button:</strong> After gating a subset of
cells, information about the newly-created gate will be displayed in a
scrollable, interactive data table below the plots.</li>
</ol>
</div>
<div class="section level4">
<h4 id="additional-features">Additional Features<a class="anchor" aria-label="anchor" href="#additional-features"></a>
</h4>
<ul>
<li>Users can click on previous gates from the data table to view the
feature scatter plot of that cell population. Using this feature, users
can also revert back to a previously created gate for further subsetting
and gating.</li>
<li>To permanently delete all gating information in the data table,
click the “Clear All Data” button.</li>
<li>To rename the selected cell subset after gating, double-click in the
corresponding area of the data, then press the Tab key or click outside
of the table.</li>
</ul>
<p><img src="gate_datatable_renaming.png" width="100%" style="display: block; margin: auto;"></p>
<ul>
<li>To reset the feature scatter plot and view the entire input cell
population, click the “Reset Scatter” button.</li>
<li>To view detailed gating data, users can download this data in a
variety of formats listed at the bottom of the gating page.</li>
<li>To reset the colors of the cells in the reduction plot, double-click
anywhere in the feature scatter plot.</li>
</ul>
</div>
<div class="section level4">
<h4 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h4>
<p>In the following example, we demonstrate a 2-layer gate to isolate
cells with specific levels of CD11b-1 and CD45-1 (mixture of myeloid and
lymphoid cells) from the whole cell population, followed by the
isolation of CD8+ CD4- cells from this first subset to view the CD8+
T-cell subpopulation:</p>
<ol style="list-style-type: decimal">
<li>Select cells with relatively low levels of CD11b-1 and CD45-1.</li>
</ol>
<p><img src="gating_CD8_T_Cells.png" width="100%" style="display: block; margin: auto;"></p>
<ol start="2" style="list-style-type: decimal">
<li>From the cells in the first gate, select cells with low levels of
CD4 and high levels of CD8.</li>
</ol>
<p><img src="gating_CD8_T_Cells2.png" width="100%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level3">
<h3 id="back-gating">Back-Gating<a class="anchor" aria-label="anchor" href="#back-gating"></a>
</h3>
<p>Users can back-gate on a selection of cells in a reduction plot and
highlight them in feature space. On the back-gating page, users can
highlight cells in feature space from a “labels-first” or “top-down”
workflow. For example, users can select a range of cells from the
reduction plot (e.g. UMAP, PCA etc.) and locate them in the feature
scatter plot for more extensive data exploration.</p>
<div class="section level4">
<h4 id="general-workflow-1">General Workflow<a class="anchor" aria-label="anchor" href="#general-workflow-1"></a>
</h4>
<p>Plot settings, cell population selection and naming methods are
analogous to the gating page. To reset the colors of the cells in the
feature scatter plot, double-click anywhere in the reduction plot.</p>
</div>
<div class="section level4">
<h4 id="example-1">Example<a class="anchor" aria-label="anchor" href="#example-1"></a>
</h4>
<p>In the following example, we back-gate on a selection of B-cells in
the WNN.UMAP reduction plot. These selected cells are then highlighted
in black in the feature scatter plot.</p>
<p><img src="backgate-b-cells.png" width="100%" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] CITEViz_0.1      BiocStyle_2.28.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] tidyselect_1.2.0            viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt;   [3] golem_0.4.1                 dplyr_1.1.3                </span></span>
<span><span class="co">#&gt;   [5] bitops_1.0-7                lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;   [7] fastmap_1.1.1               SingleCellExperiment_1.22.0</span></span>
<span><span class="co">#&gt;   [9] RCurl_1.98-1.12             promises_1.2.1             </span></span>
<span><span class="co">#&gt;  [11] digest_0.6.33               mime_0.12                  </span></span>
<span><span class="co">#&gt;  [13] lifecycle_1.0.3             ellipsis_0.3.2             </span></span>
<span><span class="co">#&gt;  [15] SeuratObject_4.1.4          magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [17] compiler_4.3.1              config_0.3.2               </span></span>
<span><span class="co">#&gt;  [19] rlang_1.1.1                 sass_0.4.7                 </span></span>
<span><span class="co">#&gt;  [21] tools_4.3.1                 utf8_1.2.3                 </span></span>
<span><span class="co">#&gt;  [23] yaml_2.3.7                  data.table_1.14.8          </span></span>
<span><span class="co">#&gt;  [25] knitr_1.44                  S4Arrays_1.0.6             </span></span>
<span><span class="co">#&gt;  [27] htmlwidgets_1.6.2           sp_2.1-1                   </span></span>
<span><span class="co">#&gt;  [29] DelayedArray_0.26.7         abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [31] purrr_1.0.2                 BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt;  [33] desc_1.4.2                  grid_4.3.1                 </span></span>
<span><span class="co">#&gt;  [35] stats4_4.3.1                fansi_1.0.5                </span></span>
<span><span class="co">#&gt;  [37] xtable_1.8-4                colorspace_2.1-0           </span></span>
<span><span class="co">#&gt;  [39] future_1.33.0               progressr_0.14.0           </span></span>
<span><span class="co">#&gt;  [41] ggplot2_3.4.4               globals_0.16.2             </span></span>
<span><span class="co">#&gt;  [43] scales_1.2.1                SummarizedExperiment_1.30.2</span></span>
<span><span class="co">#&gt;  [45] cli_3.6.1                   crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [47] rmarkdown_2.25              ragg_1.2.6                 </span></span>
<span><span class="co">#&gt;  [49] generics_0.1.3              future.apply_1.11.0        </span></span>
<span><span class="co">#&gt;  [51] httr_1.4.7                  cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [53] stringr_1.5.0               zlibbioc_1.46.0            </span></span>
<span><span class="co">#&gt;  [55] parallel_4.3.1              BiocManager_1.30.22        </span></span>
<span><span class="co">#&gt;  [57] XVector_0.40.0              matrixStats_1.0.0          </span></span>
<span><span class="co">#&gt;  [59] vctrs_0.6.4                 Matrix_1.6-1.1             </span></span>
<span><span class="co">#&gt;  [61] jsonlite_1.8.7              bookdown_0.36              </span></span>
<span><span class="co">#&gt;  [63] IRanges_2.34.1              S4Vectors_0.38.2           </span></span>
<span><span class="co">#&gt;  [65] listenv_0.9.0               systemfonts_1.0.5          </span></span>
<span><span class="co">#&gt;  [67] attempt_0.3.1               plotly_4.10.2              </span></span>
<span><span class="co">#&gt;  [69] tidyr_1.3.0                 jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [71] glue_1.6.2                  parallelly_1.36.0          </span></span>
<span><span class="co">#&gt;  [73] pkgdown_2.0.7               codetools_0.2-19           </span></span>
<span><span class="co">#&gt;  [75] DT_0.30                     stringi_1.7.12             </span></span>
<span><span class="co">#&gt;  [77] gtable_0.3.4                later_1.3.1                </span></span>
<span><span class="co">#&gt;  [79] GenomeInfoDb_1.36.4         GenomicRanges_1.52.1       </span></span>
<span><span class="co">#&gt;  [81] munsell_0.5.0               tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [83] pillar_1.9.0                htmltools_0.5.6.1          </span></span>
<span><span class="co">#&gt;  [85] GenomeInfoDbData_1.2.10     R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [87] textshaping_0.3.7           rprojroot_2.0.3            </span></span>
<span><span class="co">#&gt;  [89] Biobase_2.60.0              evaluate_0.22              </span></span>
<span><span class="co">#&gt;  [91] shiny_1.7.5.1               lattice_0.21-8             </span></span>
<span><span class="co">#&gt;  [93] memoise_2.0.1               httpuv_1.6.11              </span></span>
<span><span class="co">#&gt;  [95] bslib_0.5.1                 Rcpp_1.0.11                </span></span>
<span><span class="co">#&gt;  [97] xfun_0.40                   fs_1.6.3                   </span></span>
<span><span class="co">#&gt;  [99] MatrixGenerics_1.12.3       pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Garth Kong, Thai Nguyen, Wesley Rosales, Anjali Panikar, John Cheney.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
